packages/contracts-periphery/contracts/universal/drippie/Drippie.sol

```diff
@@ -0,0 +1,244 @@
+     */
+    struct DripConfig {
+        uint256 interval;
+        IDripCheck dripcheck;
+        bytes checkparams;
+        DripAction[] actions;
+    }
+
+    /**
+     * Represents the state of an active drip.
+     */
+    struct DripState {
+        DripStatus status;
+        DripConfig config;
+        uint256 last;
+    }
+
+    /**
+     * Emitted when a new drip is created.
+     */
+    event DripCreated(string indexed name, DripConfig config);
+
+    /**
+     * Emitted when a drip status is updated.
+     */
+    event DripStatusUpdated(string indexed name, DripStatus status);
+
+    /**
+     * Emitted when a drip is executed.
+     */
+    event DripExecuted(string indexed name, address indexed executor, uint256 timestamp);
+
+    /**
+     * Maps from drip names to drip states.
+     */
+    mapping(string => DripState) public drips;
+
+    /**
+     * @param _owner Initial contract owner.
+     */
+    constructor(address _owner) AssetReceiver(_owner) {}
+
+    /**
+     * Creates a new drip with the given name and configuration. Once created, drips cannot be
+     * modified in any way (this is a security measure). If you want to update a drip, simply pause
+     * (and potentially archive) the existing drip and create a new one.
+     *
+     * @param _name Name of the drip.
+     * @param _config Configuration for the drip.
+     */
+    function create(string memory _name, DripConfig memory _config) public onlyOwner {
+        // Make sure this drip doesn't already exist. We *must* guarantee that no other function
+        // will ever set the status of a drip back to NONE after it's been created. This is why
+        // archival is a separate status.
+        require(
+            drips[_name].status == DripStatus.NONE,
+            "Drippie: drip with that name already exists"
+        );
+
+        // We initialize this way because Solidity won't let us copy arrays into storage yet.
+        DripState storage state = drips[_name];
+        state.status = DripStatus.PAUSED;
+        state.config.interval = _config.interval;
+        state.config.dripcheck = _config.dripcheck;
+        state.config.checkparams = _config.checkparams;
+
+        // Solidity doesn't let us copy arrays into storage, so we push each array one by one.
+        for (uint256 i = 0; i < _config.actions.length; i++) {
+            state.config.actions.push(_config.actions[i]);
+        }
+
+        // Tell the world!
+        emit DripCreated(_name, _config);
+    }
+
+    /**
+     * Sets the status for a given drip. The behavior of this function depends on the status that
+     * the user is trying to set. A drip can always move between ACTIVE and PAUSED, but it can
+     * never move back to NONE and once ARCHIVED, it can never move back to ACTIVE or PAUSED.
+     *
+     * @param _name Name of the drip to update.
+     * @param _status New drip status.
+     */
+    function status(string memory _name, DripStatus _status) public onlyOwner {
```

Comment:
I'll come up with a better name for this one

Reply:
Gonna keep `status`, I don't think this will be a widely used contract and I like the shortness of it since every other function in this contract is just a single word. Will count this as my monthly "cuz I can, stylistically" allocation, lol.